swarm/testutil/http.go

```diff
@@ -49,24 +54,86 @@ func NewTestSwarmServer(t *testing.T) *TestSwarmServer {
 		ChunkStore: localStore,
 	}
 	dpa.Start()
-	a := api.NewApi(dpa, nil)
+
+	// mutable resources test setup
+	resourceDir, err := ioutil.TempDir("", "swarm-resource-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ipcPath := filepath.Join(resourceDir, "test.ipc")
+	ipcl, err := rpc.CreateIPCListener(ipcPath)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Cleanup needs to happen before calling `t.Fatal`, otherwise the temporary directories will leak.